### PR TITLE
[adios2] add port

### PR DESCRIPTION
**Pull request by Neumann-A**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html) (Hopefully)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [~] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context. (Probably not if heuristic didn't get fixed.)
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/adios2/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      ADIOS2_USE_DAOS
+      ADIOS2_USE_DataMan
+      ADIOS2_USE_SysVShMem
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
+
+vcpkg_copy_tools(TOOL_NAMES adios2_iotest adios2_reorganize adios2_reorganize_mpi bpls AUTO_CLEAN)
```
Comment (Neumann-A) [inline ports/adios2/portfile.cmake:44]:
```suggestion
vcpkg_copy_tools(TOOL_NAMES adios2_reorganize adios2_reorganize_mpi bpls AUTO_CLEAN)
```

**Comment by Neumann-A** [comment]
of course it fails because I build it with all features enabled .....

File: ports/adios2/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        mpi     ADIOS2_USE_MPI
+        cuda    ADIOS2_USE_CUDA
+        python  ADIOS2_USE_Python # requires numpy / mpi4py; so not exposed in the manifest yet
+)
+
+set(disabled_options "")
+list(APPEND disabled_options ZFP SZ LIBPRESSIO MGARD DAOS DataMan DataSpaces MHS SST BP5 IME Fortran SysVShMem Profiling )
```
Comment (Adela0814) [inline ports/adios2/portfile.cmake:17]:
```suggestion
list(APPEND disabled_options ZFP SZ LIBPRESSIO MGARD DAOS DataMan DataSpaces MHS SST BP5 IME Fortran SysVShMem Profiling)
```

File: ports/adios2/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+  file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/adios2/toolkit/sst/dp" "${CURRENT_PACKAGES_DIR}/include/adios2/toolkit/sst/util")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME "copyright")
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```
Comment (Adela0814) [inline ports/adios2/portfile.cmake:62]: Please add a new line at the end of file.

File: ports/adios2/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+  file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/adios2/toolkit/sst/dp" "${CURRENT_PACKAGES_DIR}/include/adios2/toolkit/sst/util")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME "copyright")
```
Comment (Adela0814) [inline ports/adios2/portfile.cmake:61]: Please use `vcpkg_install_copyright`.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**